### PR TITLE
Remove collect_dict_from_file function from general.py; Rename collect_data_from_yaml_or_dict

### PR DESCRIPTION
**Pull request by GernotMaier**
Follow suggestion from @VictorBarbosaMartins during the PR #763 discussion to rename `collect_data_from_yaml_or_dict` to `collect_data_from_file_or_dict`. This reflects better the added functionality of reading also json files.

Remove function `collect_dict_from_file` from general.py, as it provided almost the same functionality as `collect_data_from_yaml_or_dict`.

Removes also the temporary url forgotten in tests/unit_tests/utils/test_general.py (from PR 763).

### CODE REVIEW

File: simtools/utils/general.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def collect_data_from_http(url):
     return data
 
 
-def collect_data_from_yaml_or_dict(in_yaml, in_dict, allow_empty=False):
+def collect_data_from_file_or_dict(in_yaml, in_dict, allow_empty=False):
```
Comment (VictorBarbosaMartins) [inline simtools/utils/general.py:372]: I suggest replacing the argument `in_yaml` by `file` or `file_name` as the file is not necessarily a yaml file anymore.

**Review by VictorBarbosaMartins — CHANGES_REQUESTED** [review]
Thanks for these changes. Just one comment on the argument of `collect_data_from_file_or_dict` which should also be changed, otherwise all good

File: simtools/utils/general.py (hunk 1 above)
Comment (GernotMaier) [reply simtools/utils/general.py]: Good point - changed it to `file_name`.

**Review by VictorBarbosaMartins — APPROVED** [review]
All good. let's merge when the tests pass.

**Pull request merged**